scipy/stats/tests/test_stats.py

```diff
@@ -509,10 +517,10 @@ def test_length_two_neg1(self, xp):
         r, p = res
         one = xp.asarray(1.)
         xp_assert_equal(r, -one)
-        xp_assert_equal(p, one)
+        xp_assert_equal(p, one, check_0d=False)
         low, high = res.confidence_interval()
         xp_assert_equal(low, -one)
-        xp_assert_equal(high, one)
+        xp_assert_equal(high, one, check_0d=False)
```

Comment:
While I didn't _fully_ optimize each addition of `, check_0d=False`, especially in very long tests, I did try to keep it minimal. This surfaced some interesting cases like this one, where apparently there appears to be some sensitivity to positive/negative values?

Looking at the values in question:
```
>>> res
PearsonRResult(statistic=np.float64(-1.0), pvalue=np.float64(1.0))
>>> res.confidence_interval()
ConfidenceInterval(low=np.float64(-1.0), high=np.float64(1.0))
```
I consider this a defect in the current implementation of this PR, though I don't know yet where it creeps in.

Reply:
Ahhhhhhhhhhh, already the `-one` is enough to transform a 0d array into a scalar.
```
>>> one = np.asarray(1.)
>>> -one
np.float64(-1.0)
```
Geez Louise 😅